tests/units/Ticket.php

```diff
@@ -1077,4 +1077,174 @@ public function testUpdateFollowup() {
+         )->isEqualTo($user['pos']);
+
+         // TicketValidation
+         $val = new \TicketValidation();
+         $this->integer(
+            (int)$val->add([
+               'tickets_id'   => $tickets_id,
+               'comment_submission'      => 'A simple validation',
+               'users_id_validate' => 5, // normal
+               'status' => 2
+            ])
+         )->isGreaterThan(0);
+
+         $this->integer(
+            (int)$val->fields['timeline_position']
+         )->isEqualTo($user['pos']);
+      }
+   }
+
+   protected function _testGetTimelinePositionSolution($tlp, $tickets_id) {
+      foreach ($tlp as $users_name => $user) {
+         $auth = new \Auth();
+         $this->boolean((boolean)$auth->login($users_name, $user['pass'], true))->isTrue();
+         $uid = getItemByTypeName('User', $users_name, true);
+
+         // Ticket Solution
+         $tkt = new \Ticket();
+         $this->boolean(
+            (boolean)$tkt->update([
+               'id'   => $tickets_id,
+               'solution'      => 'A simple solution from '.$users_name
+            ])
+         )->isEqualto(true);
+
+         $this->integer(
+            (int)$tkt->getTimelinePosition($tickets_id, 'Solution', $uid)
+         )->isEqualTo($user['pos']);
+      }
+   }
+
+   function testGetTimelinePosition() {
+      // login glpi
+      $auth = new \Auth();
+      $this->boolean((boolean)$auth->login('glpi', 'glpi', true))->isTrue();
+
+      // create ticket
+      // with post-only as requester
+      // tech as assigned to
+      // normal as observer
+      $ticket = new \Ticket();
+      $this->integer((int)$ticket->add([
+            'name'                => 'ticket title',
+            'description'         => 'a description',
+            'content'             => '',
+            '_users_id_requester' => '3', // post-only
+            '_users_id_observer'  => '5', // normal
+            '_users_id_assign'    => ['4', '5'] // tech and normal
+      ] ))->isGreaterThan(0);
+
+      $tlp = [
+         'glpi'      => ['pass' => 'glpi',     'pos' => \CommonITILObject::TIMELINE_LEFT],
+         'post-only' => ['pass' => 'postonly', 'pos' => \CommonITILObject::TIMELINE_LEFT],
+         'tech'      => ['pass' => 'tech',     'pos' => \CommonITILObject::TIMELINE_RIGHT],
+         'normal'    => ['pass' => 'normal',   'pos' => \CommonITILObject::TIMELINE_RIGHT]
+      ];
+
+      $this->_testGetTimelinePosition($tlp, $ticket->getID());
+
+      // Solution timeline tests
+      $tlp = [
+         'tech'      => ['pass' => 'tech',     'pos' => \CommonITILObject::TIMELINE_RIGHT]
+      ];
+
+      $this->_testGetTimelinePositionSolution($tlp, $ticket->getID());
+
+      return $ticket->getID();
+   }
+
+   function testGetTimelineItems() {
+
+      $tkt_id = $this->testGetTimelinePosition();
+
+      // login glpi
+      $auth = new \Auth();
+      $this->boolean((boolean)$auth->login('glpi', 'glpi', true))->isTrue();
```

Comment:
``$this->login()``

Reply:
DbTestCase::login doesn't permit login with 'glpi' user, but only with TU_USER

Reply:
same as above